### DISCUSSION TITLE
Easiest way to get polygons back after triangulated meshes are sent to manifold

### DISCUSSION THREAD

**Discussion by yetigit** [elalish/manifold#207] [Q&A]
what would be the most efficient way to triangulate (the polygons intersecting or the whole meshes ? ) 
, send it to manifold for some csg ops and do some post-processing to recover quads and other n-gons ?

**Comment by makc** [comment]
there is no promise that the intersection will be planar. and then if it is curved, there is no way to triangulate it (or, rather, there are many ways, resulting in different shapes)

**Comment by elalish** [comment]
@yetigit This is one of several uses for [`MeshRelation`](https://manifoldcad.org/docs/html/classmanifold_1_1_manifold.html#a45ea90568e3c9072089e9854cd51a02f). Any coplanar triangles in the input mesh will be automatically considered as coming from the same face (for the purposes of decimation and relation), which should make recovering input quads pretty straightforward, assuming they are planar. The behavior can be controlled by specifying [`properties`](https://manifoldcad.org/docs/html/classmanifold_1_1_manifold.html#a6fcdceb0502c33d24913c6936914dd78) on the input mesh.

**Comment by stewartoallen** [comment]
@elalish I'm also interested in sometimes having Manifold just return the co-planar polygon groups. The links your reply are broken.

**Reply by elalish** [reply]
Thanks, I just fixed the links. Does that help? What is your use case?

**Reply by stewartoallen** [reply]
I'm just catching up on discussions and issues. Haven't had a chance to download and build yet. My use case will be in-browser Javascript backed by WASM. So as long as the API tracks, likely.
